tests/test_pkg.py

```diff
@@ -1,35 +1,61 @@
+"""Check the content of source code, source distribution and binary distribution."""
+
 from __future__ import annotations
 
 import os
+import subprocess
 from glob import glob
+from pathlib import Path
 
 import pytest
 
-SRC_TXT_PATH = "src/pymatgen.egg-info/SOURCES.txt"
+SRC_DIR = Path(__file__).parent.parent
+
+
+class TestCheckDistribution:
+    def test_source_code(self):
+        """Directly check the source code in the working directory."""
+        src_text_path = SRC_DIR / "src/pymatgen.egg-info/SOURCES.txt"
+
+        check_src_txt_is_complete(SRC_DIR, src_text_path)
+
+    @pytest.mark.skip(reason="WIP")  # TODO: parameterize for bdist/sdist?
+    def test_source_distribution(self):
+        """Build the source distribution and verify its contents."""
+        # Build the source distribution in ScratchDir
+        subprocess.run(["python", "-m", "build", "--sdist"], check=True)
 
+        # Decompress
 
-@pytest.mark.skipif(
-    not os.path.isfile(SRC_TXT_PATH),
-    reason=f"{SRC_TXT_PATH=} not found. Run `pip install .` to create",
-)
-def test_egg_sources_txt_is_complete():
-    """Check that all source and data files in pymatgen/ are listed in pymatgen.egg-info/SOURCES.txt."""
+        # Verify the distribution contents
+        check_src_txt_is_complete(sdist_path, SRC_TXT_PATH)
 
-    with open(SRC_TXT_PATH, encoding="utf-8") as file:
+    @pytest.mark.skip(reason="WIP")
+    def test_binary_distribution(self):
+        """Build the binary distribution (wheels) and verify its contents."""
+
+
+def check_src_txt_is_complete(src_dir: str, src_txt_path: str) -> None:
+    """Check that all source code and data files are listed in given SOURCES.txt.
+
+    Args:
+        src_dir (str): Path to the source code directory.
+        src_txt_path (str): Path to the "SOURCES.txt" file.
+    """
+    assert os.path.isdir(src_dir)
+    assert os.path.isfile(src_txt_path)
+
+    with open(src_txt_path, encoding="utf-8") as file:
         sources = file.read()
 
-    # check that all files listed in SOURCES.txt exist
+    # Check that all files listed in "SOURCES.txt" exist
     for src_file in sources.splitlines():
         assert os.path.isfile(src_file), f"{src_file!r} does not exist!"
 
-    # check that all files in pymatgen/ are listed in SOURCES.txt
-    for ext in ("py", "json*", "yaml", "csv"):
-        for filepath in glob(f"pymatgen/**/*.{ext}", recursive=True):
-            unix_path = filepath.replace("\\", "/")
-            if unix_path.endswith("dao.py"):
```

Comment:
I believe `dao.py` is already included in sdist and bdist (not just this branch, but from [PyPI](https://pypi.org/project/pymatgen/#files))